src/runtime/hexagon/hexagon_htp.cc

```diff
@@ -35,39 +35,9 @@ namespace tvm {
 namespace runtime {
 namespace hexagon {
 
-HexagonHtp::HexagonHtp() {
-  PowerOn();
-  Acquire();
-}
-
-HexagonHtp::~HexagonHtp() {
-  Release();
-  PowerOff();
-}
-
-void HexagonHtp::PowerOn() {
-  HAP_power_request_t pwr_req;
-  int nErr;
-
-  hap_pwr_ctx_ = HAP_utils_create_context();
-  pwr_req.type = HAP_power_set_HMX;
-  pwr_req.hmx.power_up = true;
-  if ((nErr = HAP_power_set(hap_pwr_ctx_, &pwr_req))) {
-    LOG(FATAL) << "InternalError: HAP_power_set failed\n";
-  }
-}
-
-void HexagonHtp::PowerOff() {
-  HAP_power_request_t pwr_req;
-  int nErr;
+HexagonHtp::HexagonHtp() { Acquire(); }
```

Comment:
I'm working on this area, too.  It is similar, but Acquire/Release are public methods now.  And I'll be splitting out Acquire/Release to be called separately from thread_main and thread_exit.  What you have here is good - it will be a merge for me, though!